azurelinuxagent/ga/update.py

```diff
@@ -485,7 +485,7 @@ def _try_update_goal_state(self, protocol):
         try:
             max_errors_to_log = 3
 
-            protocol.update_goal_state(silent=self._update_goal_state_error_count >= max_errors_to_log)
+            protocol.client.update_goal_state(silent=self._update_goal_state_error_count >= max_errors_to_log)
```

Comment:
@narrieta On line 430 in update.py we discussed changing 
`self._try_update_goal_state(protocol)`
to 
`self._reset_goal_state()`

It looks like we sleep until try_update_goal_state returns True for success.. Should we implement similar logic for reset_goal_state?

Reply:
good point. i added a comment on _try_update_goal_state